File: HomeWorks/src/Lesson6/Heads/ToshibaHead.java
package Lesson6.Heads;

public class ToshibaHead implements IHead {
    private int price;

    public ToshibaHead (int price) {
        this.price = price;
    }

    public ToshibaHead() {
    }

    @Override
    public void speak() {
        System.out.println("Говорит голова Toshiba");
    }

    public void setPrice(int price) {
        this.price = price;
    }

    @Override
    public int getPrice() {
        return price;
    }
}

File: HomeWorkKotlin/src/Robot/Heads/ToshibaHead.kt
package Robot.Heads

class ToshibaHead(private val price_param : Int) : IHead {

    private var price = price_param
        set(value) {
            if(value > 0) field = value
        }

    override fun speak() {
        println("Машет рука Toshiba")
    }

    override fun getHeadPrice() : Int = price
}